From 5fcd385572ccf2d4a3c3030723617eaec8ae1516 Mon Sep 17 00:00:00 2001
From: Mark Reynolds <mreynolds@redhat.com>
Date: Tue, 21 Apr 2015 12:23:47 -0400
Subject: [PATCH 314/319] Ticket 48154 - abort cleanAllRUV tasks should not
 certify-all by default

Bug Description:  The current default for an abort task is to certify that
                  all the replicas receive and process the abort task.  The
                  problem is that if we are trying to abort a cleanallruv
                  task because it is "hanging" due to some replicas not
                  being reachable, the abort task will also "hang".

Fix Description:  The default for certify-all should be "no".

https://fedorahosted.org/389/ticket/48154

Reviewed by: nhosoi(Thanks!)

(cherry picked from commit 465e1059493a8dacf3caf7776824e8149e878b93)
(cherry picked from commit ba488271c51a0bd566a4f728709ff959d64a0ed3)
---
 ldap/servers/plugins/replication/repl5_replica_config.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/ldap/servers/plugins/replication/repl5_replica_config.c b/ldap/servers/plugins/replication/repl5_replica_config.c
index 94ab71b..daefc03 100644
--- a/ldap/servers/plugins/replication/repl5_replica_config.c
+++ b/ldap/servers/plugins/replication/repl5_replica_config.c
@@ -2596,7 +2596,13 @@ replica_cleanall_ruv_abort(Slapi_PBlock *pb, Slapi_Entry *e, Slapi_Entry *eAfter
             goto out;
         }
     } else {
-        certify_all = "yes";
+        /*
+         * The default should be not to certify all the replicas, because
+         * we might be trying to abort a clean task that is "hanging" due
+         * to unreachable replicas.  If the default is "yes" then the abort
+         * task will run into the same issue.
+         */
+        certify_all = "no";
     }
     /*
      *  Create payload
-- 
1.9.3

